components/Header.tsx

```diff
@@ -1,11 +1,105 @@
+
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
     return ( 
-        <header>
-            <h2>This is the Header all-mighty</h2>
-            <Link href="/">Home</Link>
-            <Link href="/works">Works</Link>
+        <header className="fixed top-0 w-full h-0">
+            <div className={`flex justify-between items-center px-5 md:px-14 h-[5.5rem] z-10 ease-in duration-200 ${(currentPath === "/" && !toggleMenu) ? (isScrolled && 'bg-project-black'): ('bg-project-black')}`}>
+                <Link onClick={setMenuState} href="/"><img src="/logo-white.png" className=" h-10 w-fit" alt="logo"></img></Link>
+                <nav className=" hidden lg:block">
+                    <ul className="flex gap-x-7 uppercase text-base items-center">
+                        <li className="relative"><Link href="/" className={`block px-4 py-3 ${isActive('/')}`}>Home<span className="absolute left-1/2 bottom-0 border-b-2 w-0 border-project-green transition-all duration-300 ease-in-out group-hover:w-full transform -translate-x-1/2"></span></Link></li>
+                        <li className="relative"><Link href="/works" className={`block px-4 py-3 ${isActive('/works')}`}>Works<span className="absolute left-1/2 bottom-0 border-b-2 w-0 border-project-green transition-all duration-300 ease-in-out group-hover:w-full transform -translate-x-1/2"></span></Link></li>
+                        <li className="relative"><Link href="/about" className={`block px-4 py-3 ${isActive('/about')}`}>About<span className="absolute left-1/2 bottom-0 border-b-2 w-0 border-project-green transition-all duration-300 ease-in-out group-hover:w-full transform -translate-x-1/2"></span></Link></li>
+                        <li className="relative"><Link href="/contact" className={`block px-4 py-3 ${isActive('/contact')}`}>Contact<span className="absolute left-1/2 bottom-0 border-b-2 w-0 border-project-green transition-all duration-300 ease-in-out group-hover:w-full transform -translate-x-1/2"></span></Link></li>
+                    </ul>
+                </nav>
+                <div onClick={setMenuState} className="block lg:hidden cursor-pointer py-2">
+                    {/* <svg className="w-10 h-fit" viewBox="0 0 38 38" fill="none" xmlns="http://www.w3.org/2000/svg">
+                        <rect className=" origin-center rotate-45" y="18" x="1" width="38" height="5" fill="white"/>
+                        <rect className=" origin-center -rotate-45" y="18" x="-1" width="38" height="5" fill="white"/>
+                    </svg> */}
+                    <svg className=" w-9 md:w-10 h-fit" viewBox="0 0 38 38" fill="none" xmlns="http://www.w3.org/2000/svg">
+                        <rect className={ ` ease-in-out duration-300 origin-center ${toggleMenu ? 'translate-y-0 translate-x-0 rotate-45' : '-translate-y-[6px] -translate-x-[1px]'}` } y="18" x="1" width="38" height="5" fill="white"/>
```

Comment:
Remove any extra spaces, for example ``{ ` ease-in-out`` should look like ``{`ease-in-out``